aptos-move/move-examples/post_mint_reveal_nft/sources/minting.move

```diff
@@ -0,0 +1,994 @@
+                // If the claimer's address is not on the public_minting_addresses table yet, it means this is the
+                // first time that this claimer mints. We will add the claimer's address and remaining amount of mints
+                // to the public_minting_addresses table.
+                if (!bucket_table::contains(&public_mint_config.public_minting_addresses, &claimer_addr)) {
+                    bucket_table::add(&mut public_mint_config.public_minting_addresses, claimer_addr, *option::borrow(&collection_config.public_mint_limit));
+                };
+                let limit = bucket_table::borrow_mut(&mut public_mint_config.public_minting_addresses, claimer_addr);
+                assert!(amount <= *limit, error::invalid_argument(EAMOUNT_EXCEEDS_MINTS_ALLOWED));
+                *limit = *limit - amount;
+            };
+            while (amount > 0) {
+                mint_source_certificate_internal(nft_claimer, public_mint_config.public_mint_price);
+                amount = amount - 1;
+            };
+        };
+    }
+
+    // Exchange a source certificate token to a destination token. This function will burn the source certificate
+    // and put a destination token in the nft_claimer's TokenStore.
+    entry fun exchange(nft_claimer: &signer, property_version: u64) acquires NFTMintConfig, CollectionConfig, RevealConfig {
+        assert!(exists<CollectionConfig>(@post_mint_reveal_nft) && exists<RevealConfig>(@post_mint_reveal_nft), error::permission_denied(ECONFIG_NOT_INITIALIZED));
+
+        let reveal_config = borrow_global<RevealConfig>(@post_mint_reveal_nft);
+
+        assert!(timestamp::now_seconds() > reveal_config.reveal_time, error::permission_denied(ECANNOT_EXCHANGE_BEFORE_REVEAL_SRARTS));
+
+        let collection_config = borrow_global_mut<CollectionConfig>(@post_mint_reveal_nft);
+        let source_collection_name = collection_config.collection_name;
+        string::append_utf8(&mut source_collection_name, b" - source collection");
+        let token_id = token::create_token_id_raw(@post_mint_reveal_nft, source_collection_name, utf8(b""), property_version);
+        assert!(token::balance_of(signer::address_of(nft_claimer), token_id) > 0, error::invalid_argument(ETOKEN_ID_NOT_FOUND));
+
+        let now = timestamp::now_microseconds();
+        let nft_mint_config = borrow_global_mut<NFTMintConfig>(@post_mint_reveal_nft);
+
+        // Assert there's still some token uris in the vector.
+        assert!(big_vector::length(&collection_config.tokens) > 0, error::resource_exhausted(ENO_ENOUGH_TOKENS_LEFT));
+
+        // Randomize which token we're assigning to the user.
+        let index = now % big_vector::length(&collection_config.tokens);
+        let bucket_index = big_vector::bucket_index(&collection_config.tokens, index);
+        let token: TokenAsset = big_vector::swap_remove(&mut collection_config.tokens, &bucket_index);
```

Comment:
nit: can't TokenAsset type be inferred here?

Reply:
yeah it can, I'll remove this